file-share-service-desktop-client/App.xaml.cs

```diff
@@ -29,11 +35,19 @@ public App()
 
         protected override IContainerExtension CreateContainerExtension()
         {
-            var containerExtension = base.CreateContainerExtension() as UnityContainerExtension;
-#if DEBUG
-            containerExtension.Instance.AddExtension(new Diagnostic());
-#endif
-            return containerExtension;
+            //            var containerExtension = base.CreateContainerExtension() as UnityContainerExtension;
+            //#if DEBUG
+            //            containerExtension.Instance.AddExtension(new Diagnostic());
+            //#endif
+            //            return containerExtension;
+
+            var serviceCollection = new ServiceCollection();
+            serviceCollection.AddLogging(loggingBuilder => loggingBuilder.AddSerilog(dispose: true));
```

Comment:
We could initialise the logger here
````
serviceCollection.AddLogging(loggingBuilder => {
                var logger = new LoggerConfiguration()
                        .ReadFrom.AppSettings()
                        .CreateLogger();
                loggingBuilder.AddSerilog(logger, dispose: true);
            });
  ````
  to keep the logger setup code together
  
  (the logger setup in `CreateShell` could then be removed)

Reply:
Thank you for review comments. Implemented this change.